mautic/mautic
# WIP: Web Push Notification update with new options and images/badge/icons upload

closes https://github.com/mautic/mautic/issues/5662

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/292
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5662
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Added new cool features to web notifications

- Setting the Icon, Image, Badge  https://documentation.onesignal.com/docs/web-push-notification-icons (badge not included - don't have api support)
- Customize action buttons https://documentation.onesignal.com/docs/action-buttons
- Time to live https://documentation.onesignal.com/docs/sending-notifications#section-time-to-live
- Priority - https://documentation.onesignal.com/reference#section-delivery 

How should it look:

![new web notification mautic](https://user-images.githubusercontent.com/462477/42929087-54bb71c0-8b39-11e8-9002-efeefd8713e0.png)

#### Steps to test this PR:

**Test DB migration**:

1.  Apply PR and run `php app/console doctrine:migrations:migrate`
2. Run  `php app/console doctrine:schema:update --dump-sql` 
3. Works If nothing is to update in DB schema

**Test feature request**: 

4.  Setup OneSignal integration https://mautic.org/docs/en/notifications/index.html  and enable web push notification
5. Go to website with Mautic tracking code from Chrome and Firefox brower
6. Confirm notification 
7. Create new web push notification in Matutic, setup both action buttons and icon and image
8. Run campaign with send  web notification action
9. Works If you received  both type of notification
- Chrome https://documentation.onesignal.com/docs/web-push-notification-icons#section-image
- Firefox https://documentation.onesignal.com/docs/web-push-notification-icons#section-icon
10. Works if you see both action button in Chrome 
https://documentation.onesignal.com/docs/action-buttons



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/5695"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
